File: Ejercicio18.java
import java.util.Scanner;

public class Ejercicio18 {
    public static void main(String[] args){
        
        int contarA=0;

        for (int i=0; i < 9; i++) {
            System.out.println("=>"+(i+1)+" ingrese un numero: ");
            Scanner sc = new Scanner(System.in);
            int s = sc.nextInt();
            
            if(s != 0) {
                contarA=contarA+s;
                System.out.println("van:"+contarA);
            }else{
                break;
            }
        }
        System.out.println("La suma es:"+contarA);
    }
    
    
}

File: Ejercicio18.cpp
#include <string>
#include <iostream>
#include <math.h>

using namespace std;

int main(){
    int contar=0;
    int n;

    for (int i = 0; i < 9; i++)
    {
        cout<<"=>"<<(i+1)<<" Ingrese un numero, si es 0 se termina"<<endl;
        cin >>n;
        if(n !=0){
            contar=contar+n;
            cout<<"La suma va en: "<<contar<<endl;
        }else{
            cout<<"A ingresado 0, finalizamos"<<endl;
            break;
        }
    
    }
    cout<<"La suma total es: "<<contar<<endl;
    return 0;
}